gdal/swig/python/scripts/gdal_calc.py

```diff
@@ -51,19 +51,25 @@
 import os.path
 import sys
 import shlex
+import string
+from collections import defaultdict
+from pathlib import Path
```

Comment:
seems to be python3 specific. We still maintain python2 compat

Reply:
hmm.. shouldn't Python2 be dead already?
I look for an alternative.

Reply:
we haven't yet officially killed it. There would be changes to do on the continuous integration side since a number of them still use it... Perhaps something for GDAL 3.3

Reply:
https://cpython-test-docs.readthedocs.io/en/latest/library/string.html?highlight=string#module-string
https://docs.python.org/2/library/collections.html#collections.defaultdict

were you referring to pathlib?
https://pypi.org/project/pathlib/
pathlib is included in python3.4 but apparently can be installed on python2. would it be acceptable or you prefer that I'll remove it?

Reply:
I'd prefer if we removed any dependence to non-core Python modules. We don't have the "formalism" currently to express such dependencies

Reply:
ok, done.